library/components/TPdf.vue

```diff
@@ -29,7 +29,12 @@
         methods: {
             downloadPdf() {
                 const url = this.page.url + window.location.search;
-                return this.downloadPdfFile(url);
+                this.downloadPdfFile(url);
+                window.ampli.reportDataExport({
+                    reportExportType: "pdf",
+                    reportName: this.page.title,
+                    reportLayerName: "null",
+                })
```

Comment:
how would you feel about moving the event into the mixin in topcoat core? I feel like it could be good for keeping all the implementation details of the tracking library inside topcoat-core, and try to keep the components free of information about the tracking library.

Reply:
We will still have some events in public though, like report dropdown one.

Reply:
This is true, but we could emit an event and deal with it in the mixin, if I'm not mistaken?

Reply:
Yes that might work though we will have to include tracking functions in all mixins depending on the component (vis_mixin, lib_mixin etc), though does make sense to include them all in core.

Reply:
@sareg0 I realised moving to mixins might not be good, because we might get more components that can fire these events and some of them will be inside vis_mixin while others will be in lib_mixin which might be hard to manage later on, what do you think?

Imagine another export component that is inside visualizations instead of library.